sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableDropPartitionSuite.scala

```diff
@@ -42,6 +42,18 @@ trait AlterTableDropPartitionSuiteBase extends command.AlterTableDropPartitionSu
       checkPartitions(t) // no partitions
     }
   }
+
+  test("SPARK-33950: refresh cache after partition dropping") {
+    withTable("t") {
+      sql(s"CREATE TABLE t (id int, part int) $defaultUsing PARTITIONED BY (part)")
+      sql("INSERT INTO t PARTITION (part=0) SELECT 0")
+      sql("INSERT INTO t PARTITION (part=1) SELECT 1")
+      sql("CACHE TABLE t")
+      checkAnswer(sql("SELECT * FROM t"), Seq(Row(0, 0), Row(1, 1)))
+      sql("ALTER TABLE t DROP PARTITION (part=0)")
+      checkAnswer(sql("SELECT * FROM t"), Seq(Row(1, 1)))
```

Comment:
nit: it will be good to verify that the table is cached after the file index issue is solved :)

Reply:
I expected your suggestion ;-). Actually, when I added the check from previous PR:
```scala
assert(spark.sharedState.cacheManager.lookupCachedData(spark.table(t)).isDefined)
```
it failed for v1 Hive external catalog. It even fails immediately after `sql("CACHE TABLE t")`.

Since this is an unified test for both In-Memory and Hive catalogs, I decided to skip the check because this fix is mostly about correctness, and `checkAnswer(sql("SELECT * FROM t"), Seq(Row(1, 1)))` checks the correct result.

Reply:
As an alternative solution, I can move this test from the base test suite to specific suites for v1 In-Memory and Hive catalogs, and add the check for In-Memory catalog.

Reply:
For some reasons `lookupCachedData` cannot find the cached hive table by the dataset `spark.table("t")`. I just checked the entire cache - is it empty or not.